fxgl-core/src/main/java/com/almasb/fxgl/core/collection/ObjectMap.java

```diff
@@ -600,21 +601,14 @@ private String toString(String separator, boolean braces) {
         K[] keyTable = this.keyTable;
         V[] valueTable = this.valueTable;
         int i = keyTable.length;
-        while (i-- > 0) {
-            K key = keyTable[i];
-            if (key == null) continue;
-            buffer.append(key);
-            buffer.append('=');
-            buffer.append(valueTable[i]);
-            break;
-        }
-        while (i-- > 0) {
-            K key = keyTable[i];
-            if (key == null) continue;
-            buffer.append(separator);
-            buffer.append(key);
-            buffer.append('=');
-            buffer.append(valueTable[i]);
+        for(int j = i; j > 0; j--) {
+            K key = keyTable[j];
+            if (key != null) {
+                buffer.append(separator);
```

Comment:
Было 2 одинаковых цикла, идущих друг за другом. По условию переменная i доходила до 0, и второй цикл не выполнялся. Но переменная separator передаётся в этот метод и по идее должна быть использована. Поэтому оставил один цикл с использованием separator.

Reply:
In the old code, `while (i-- > 0)` will check if `i > 0`, then decrement it, so `keyTable[i]` is the last element. In the new code, line `keyTable[j]` will fail with OOB because `j == i == keyTable.length`. Please can you confirm? I think we should write a test and only then refactor it, the existing code seems fragile.

Reply:
Проверил, исправил. Теперь точно работает.

Reply:
Извиняюсь, что не сразу понял логику кода. Вернул его и добавил юнит тест.